src/main/java/com/firebolt/jdbc/connection/FireboltConnectionUserPassword.java

```diff
@@ -57,6 +58,25 @@ protected void authenticate() throws SQLException {
         sessionProperties = loginProperties.toBuilder().host(engine.getEndpoint()).engine(engine.getName()).database(database).build();
     }
 
+    @Override
+    protected void validateConnectionParameters() throws SQLException {
+        String clientId = loginProperties.getPrincipal();
```

Comment:
nit: Since it's a username/password connection, let's rename clientId->username and clientSecret -> password for clarity